src/bindings/python/src/pyopenvino/graph/partial_shape.cpp

```diff
@@ -18,6 +18,17 @@
 
 namespace py = pybind11;
 
+template <typename T>
+bool compare_shape(const ov::PartialShape& a, const T& b) {
+    if (a.is_dynamic()) {
+        throw py::type_error("Cannot compare dynamic shape with " + std::string(typeid(b).name()));
```

Comment:
let's add one more test case that this message throws.
You may catch it like [here ](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_graph/test_core.py#L99)

Reply:
Done! I also changed `typeid(b).name()` for `py::str(py::type::of(b)))` so error message prints type in pythonic way like `<class 'tuple'>`. Thanks for pointing this out